### PR TITLE
Fixes #17164: Add Usermanagement endpoints in API documentation

### PR DESCRIPTION
**Pull request by ElaadF**
https://issues.rudder.io/issues/17164

### CODE REVIEW

**Comment by amousset** [comment]
needs a rebase

**Comment by ElaadF** [comment]
Commit modified

**Comment by ElaadF** [comment]
Commit modified

File: webapp/sources/api-doc/components/schemas/users.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  username:
+    type: string
+    example: John Doe
+  password:
+    type: string
+    description: this password will be hashed for you if the `isPreHashed` is set on false
+    example: passwdWillBeStoredHashed
+  role:
+    type: array
+    description: defined user's permissions
```
Comment (amousset) [inline webapp/sources/api-doc/components/schemas/users.yml:20]:
```suggestion
    description: Defined user's permissions
```

File: webapp/sources/api-doc/openapi.src.yml [hunk 2]
```diff
@@ -66,7 +66,25 @@ tags:
 
 
       Manage relays.
+  - name: "🧩 User Management"
+    description: >-
+      **Requires that the `scale-out-relay` plugin is installed on the server.**
```
Comment (amousset) [inline webapp/sources/api-doc/openapi.src.yml:71]:
```suggestion
      **Requires that the `user-management` plugin is installed on the server.**
```

File: webapp/sources/api-doc/openapi.src.yml [hunk 3]
```diff
@@ -66,7 +66,25 @@ tags:
...
+  "/usermanagement/users":
+    $ref: paths/usermanagement/all.yml
+  "/usermanagement/{username}":
+    $ref: paths/usermanagement/id.yml
+  "/usermanagement/update/{username}":
+    $ref: paths/usermanagement/update.yml
+  "/usermanagement/roles":
+    $ref: paths/usermanagement/roles.yml
+  "/usermanagement/users/reload":
+     $ref: paths/usermanagement/reload.yml
```
Comment (amousset) [inline webapp/sources/api-doc/openapi.src.yml:85]:
```suggestion
    $ref: paths/usermanagement/reload.yml
```

File: webapp/sources/api-doc/openapi.src.yml [hunk 4]
```diff
@@ -66,7 +66,25 @@ tags:
...
+  "/usermanagement":
+     $ref: paths/usermanagement/add.yml
```
Comment (amousset) [inline webapp/sources/api-doc/openapi.src.yml:87]:
```suggestion
    $ref: paths/usermanagement/add.yml
```

File: webapp/sources/api-doc/paths/usermanagement/all.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2020 Normation SAS
+get:
+  summary: List all users
+  description: Get the list of all present users and their permissions
+  operationId: getUserInfo
+  responses:
+    "200":
+      description: Users informations
```
Comment (amousset) [inline webapp/sources/api-doc/paths/usermanagement/all.yml:9]:
```suggestion
      description: Users information
```

File: webapp/sources/api-doc/paths/usermanagement/reload.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2020 Normation SAS
+get:
+  summary: Reload user
+  description: Reload the users from the file system, in the configuration file
+  operationId: reloadUserConf
+  responses:
+    "200":
+      description: Reload informations
```
Comment (amousset) [inline webapp/sources/api-doc/paths/usermanagement/reload.yml:9]:
```suggestion
      description: Reload information
```

**Pull request merged**